### ISSUE TITLE
Google apps for business

### ISSUE THREAD

**Issue by dalcris** [swiftmailer/swiftmailer#587]
I using Swift Mailer on localhost and live server using a SMTP connection (no-reply@isportsnational.com). It's works fine, but not sending emails to email address created with "Google apps for business". For example, I want to send an email to: shelly@ipagescanada.com. The domain 'ipagescanada.com' have the following MX records: ```alt1.aspmx.l.google.com, aspmx.l.google.com, aspmx2.googlemail.com, aspmx3.googlemail.com, aspmx4.googlemail.com, alt2.aspmx.l.google.com, aspmx5.googlemail.com``` 
The email address "shelly@ipagescanada.com" is real. But I cannot send emails to this address. This is the error log: 
```++ Starting Swift_SmtpTransport << 220 ca7.design2web.ca ESMTP Postfix >> EHLO isportsnational.tst << 250-ca7.design2web.ca 250-PIPELINING 250-SIZE 41943040 250-ETRN 250-STARTTLS 250-ENHANCEDSTATUSCODES 250-8BITMIME 250 DSN >> STARTTLS << 220 2.0.0 Ready to start TLS >> EHLO isportsnational.tst << 250-ca7.design2web.ca 250-PIPELINING 250-SIZE 41943040 250-ETRN 250-AUTH LOGIN PLAIN DIGEST-MD5 CRAM-MD5 250-ENHANCEDSTATUSCODES 250-8BITMIME 250 DSN >> AUTH CRAM-MD5 << 334 PDExODM0Njc3MzAuMTU2MjExODVAY2E3LmRlc2lnbjJ3ZWIuY2E+ >> bm8tcmVwbHlAaXNwb3J0c25hdGlvbmFsLmNvbSBiMjI4Y2ZlYTgyY2EwZTI3YmQ1YTRkM2JhZWY3MTFlZg== << 235 2.7.0 Authentication successful ++ Swift_SmtpTransport started >> MAIL FROM: <bounces@isportsnational.com> << 250 2.1.0 Ok >> RCPT TO: <shelly@ipagescanada.com> << 550 5.1.1 <shelly@ipagescanada.com>: Recipient address rejected: User unknown in virtual mailbox table !! Expected response code 250/251/252 but got code "550", with message "550 5.1.1 <shelly@ipagescanada.com>: Recipient address rejected: User unknown in virtual mailbox table " >> RSET << 250 2.0.0 Ok ```
My code, for reference:
``` Swift_Preferences::getInstance()->setCharset($charset);
        $transport = Swift_SmtpTransport::newInstance()
                ->setHost($smtp_host)
                ->setPort($smtp_port)
                ->setEncryption($smtp_ecryption)
                ->setUsername($smtp_user)
                ->setPassword($smtp_pass);
        $mailer = Swift_Mailer::newInstance($transport);
        $logger = new Swift_Plugins_Loggers_ArrayLogger();
        $mailer->registerPlugin(new Swift_Plugins_LoggerPlugin($logger));
        $message = Swift_Message::newInstance()
                ->setSubject('=?utf-8?Q?=F0=9F=94=91?= Password reset request')
                ->setFrom(array('no-reply@isportsnational.com' => 'iSportsNational.com'))
                ->setTo(array($email))
                ->setReturnPath('bounces@isportsnational.com')
                ->setBody($body_html, 'text/html')
                ->addPart($body_text, 'text/plain')
                ->setPriority(2);
        $result = $mailer->send($message);
        FB::log($logger->dump());

**Comment by xdecock** [comment]
Tldr-summary : not a swiftmailer bug, should be on mailing list as usual

Are you hosted on the same server with some kind of webmin / ispconfig /
... ?

If so, this is a simple variation of rtfm, you need to configure your
server not to handle emails for those destinations.

Hope this helps,
it's not a bug in swiftmailer.

**Comment by dalcris** [comment]
I have no rights to change something on server. I can manage only my application. And, I cannot put email restriction to my customers. I need all email addresses to be handled. You said that is not a bug. If you think that is normal that your library to handle some email adresses from a list, and don't  handle other (real) email adresses, for unclear (for me) reasons, you are right... But think about these: the unhandled addresses are used for registration, communications, payments confirmation, to receive orders and so on... I reported this issue to help you make your application better. I really apreciated your job. But in these conditions, I'm afraid that is not usable for me. Thank you!

**Comment by xdecock** [comment]
The issue is not in swiftmailer, the server on which your app runs is
misconfigured, in more words, the server is considering himself as being
the final mailserver for those addresses, as such, it does not look at mx
records and only check if the email address exists.

As it does not exists (you don't host mail there) it returns a 5xx error,
which, given the misconfiguration of the server is correct.

You'll have the exact same problem with the mail() function or phpmail, or
any other system, (i strongly encourage you to check).

I'll gladly check with your hosting provider, but, it's not a library bug :)

If your  hosting provider can't reconfigure this, my recommendation is to
use a professional hosting provider, because this error is way too common.

Regards,

Xavier.

On Thu, Feb 26, 2015 at 6:58 AM, dalcris <notifications@github.com> wrote:

> I have no rights to change something on server. I can manage only my
> application. And, I cannot put email restriction to my customers. I need
> all email addresses to be handled. You said that is not a bug. If you think
> that is normal that your library to handle some email adresses from a list,
> and don't handle other (real) email adresses, for unclear (for me) reasons,
> you are right... But think about these: the unhandled addresses are used
> for registration, communications, payments confirmation, to receive orders
> and so on... I reported this issue to help you make your application
> better. I really apreciated your job. But in these conditions, I'm afraid
> that is not usable for me. Thank you!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/swiftmailer/swiftmailer/issues/587#issuecomment-76126684>
> .
>



-- 
Xavier De Cock
GPG Fingerprint: 93CA EE3F 9F57 5BE1 AE4A  794D 3C74 CA9E E7A5 0C1B
GPG Id: 0xE7A50C1B

**Comment by dalcris** [comment]
Hmm ... You had right ... everything is messed on my server ... I tried to contact my web hosting service provider, but they answered me that does not offers support for such problems and recommended me to use mailgun.com or critsend.com to handle emails. The server that I use is a VPS (Ubuntu 14.04). On this server are installed servers: Postfix, Dovecot and Sendmail ... but unfortunately they are not configured properly. If I will provide you root access to SSH / Webmin, you think can you investigate the problem and could give me some practical solutions to remedy them? I partially solved the problem with sending emails, also with SwiftMailer, changing the transport: if SMTP fails, it change with Sendmail, and if it fails, it change with Mail. The problem is that I can not use my SMTP server for handle emails, and I have to use an external server. For web server credentials, contact me at: crisalin@aka4web.com . Thank you!
